src/internal/Internal.hs

```diff
@@ -0,0 +1,19 @@
+{-# OPTIONS_GHC -fno-warn-unused-imports #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- This module hierarchy provides types and functions that are not intended to
+-- be part of the public API.
+--
+-- Types and functions defined herein are not guaranteed to be forwards or
+-- backwards compatible across different versions of the library.
```

Comment:
We could actually do without this module, but I thought it useful to include, as it allows us to provide this cautionary message that covers all submodules.